client/src/utils/use-auth.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+
+// code inspired from https://usehooks.com/useAuth/
+import * as React from 'react';
```

Comment:
I think `import *` isn't super good, maybe if there's a way to import exactly what we want

Reply:
flow also doesn't like